### PR TITLE
Enforce `uninlined_format_args` for the workspace

### LINKED ISSUE #9064: add `uninlined_format_args`
Oh nice! If you'd like feel free to add this lint [here](https://github.com/bytecodealliance/wasmtime/blob/ba864e987ef1ab87c439ca6b396264547d2425e1/Cargo.toml#L174-L182) which would be a ratchet for the whole workspace. Probably not all of `pedantic` but this specific lint I think would be a reasonable candidate to add.

_Originally posted by @alexcrichton in https://github.com/bytecodealliance/wasmtime/issues/9047#issuecomment-2263425790_

### PR DESCRIPTION
**Pull request by hamirmahal**
closes #9064.

### CODE REVIEW

**Review by hamirmahal — COMMENTED** [review]
I can fix [the `/home/runner/work/wasmtime/wasmtime/target/debug/build/cranelift-codegen-d924ecd4c940ea8b/out/inst_builder.rs` failures](https://github.com/bytecodealliance/wasmtime/actions/runs/10220802939/job/28281941147?pr=9065#step:15:365) locally, but they're not in source control.

**Comment by elliottt** [comment]
> I can fix [the `/home/runner/work/wasmtime/wasmtime/target/debug/build/cranelift-codegen-d924ecd4c940ea8b/out/inst_builder.rs` failures](https://github.com/bytecodealliance/wasmtime/actions/runs/10220802939/job/28281941147?pr=9065#step:15:365) locally, but they're not in source control.

Those are generated during the build, so we should figure out a different approach for fixing the errors. Perhaps we can emit an allow annotation when generating them?

**Comment by alexcrichton** [comment]
Similar `#[allow]` annotations look [like this](https://github.com/bytecodealliance/wasmtime/blob/dbc503f3fe1864e89f4f4a4e5739b3441fb4cbc0/cranelift/codegen/src/opts/generated_code.rs#L9) and there might be some other locations as well.

File: tests/all/globals.rs [hunk 1]
```diff
@@ -145,7 +145,7 @@ fn i31ref_global_get() -> Result<()> {
             let global = Global::new(
                 &mut store,
                 GlobalType::new(ValType::I31REF, mutability),
-                val.clone(),
```
Comment (alexcrichton) [inline tests/all/globals.rs:148]: Were these clones being removed part of a separate clippy lint that was firing? Perhaps another good candidate to enable at the workspace level? (find to do in a separate PR)

**Review by alexcrichton — APPROVED** [review]
Thank you again for your work on this!

**Comment by hamirmahal** [comment]
You're welcome!

File: tests/all/globals.rs (hunk 1 above)
Comment (hamirmahal) [reply tests/all/globals.rs:148]: I believe so, yes. I might be able to get to that this weekend.

Comment (hamirmahal) [reply tests/all/globals.rs:148]: I took a look, and it seems like `Cargo.toml` already warns about `clone_on_copy`.

```
[workspace.lints.clippy]
# The default set of lints in Clippy is viewed as "too noisy" right now so
# they're all turned off by default. Selective lints are then enabled below as
# necessary.
all = { level = 'allow', priority = -1 }
clone_on_copy = 'warn'
map_clone = 'warn'
uninlined_format_args = 'warn'
unnecessary_to_owned = 'warn'
manual_strip = 'warn'
```
